tests/test_vmec.py

```diff
@@ -1278,6 +1280,487 @@ def test_vmec_boundary_subspace(DummyStellarator):
+                    quant_from_booz = quant_trans.transform(quant_mn_desc_basis)
+
+            np.testing.assert_allclose(
+                quant_from_booz,
+                data[name],
+                rtol=1e-3,
+                atol=1e-7,
+                err_msg=f"{name} at surf index {surf_index}",
+            )
+
+
+@pytest.mark.slow
+@pytest.mark.unit
+def test_make_boozmn_output_against_hidden_symmetries_booz_xform(TmpDir):
+    """Test that booz_xform-style outputs compare well against C++ implementation."""
+    # testing against https://github.com/hiddenSymmetries/booz_xform/tree/main
+    # commit 881907058ece03
+    # load in precise_QA equilibrium
+    eq = get("precise_QA")
+    output_path = str(TmpDir.join("boozmn_out.nc"))
+
+    boozer_res_M = 25
+    boozer_res_N = 25
+
+    # compare against a 100 surface Mboz=Nboz=40 run of precise QA
+    # with the hidden symmetries C++ booz_xform implementation
+    # (ran on a wout created with VMECIO.save of precise QA example with 100 surfs
+    # and Mnyq = Nnyq = 30)
+    surfs = 100
+    Cpp_booz_output_path = (
+        f"./tests/inputs/boozmn_{surfs}_surfs_precise_QA"
+        + f"_sims_booz_{boozer_res_M}.nc"
+    )
+
+    # Use DESC to calculate the boozer harmonics and create a booz_xform style .nc file
+    make_boozmn_output(
+        eq,
+        output_path,
+        surfs=surfs,
+        verbose=0,
+        M_booz=boozer_res_M,
+        N_booz=boozer_res_N,
+    )
+
+    # load in the .nc file
+    file = Dataset(output_path, mode="r")
+
+    R_mnc = file.variables["rmnc_b"][:].filled()
+    Z_mns = file.variables["zmns_b"][:].filled()
+    B_mnc = file.variables["bmnc_b"][:].filled()
+    nu_mns = file.variables["pmns_b"][:].filled()
+    g_mnc = file.variables["gmn_b"][:].filled()
+
+    quantities = [R_mnc, Z_mns, B_mnc, g_mnc, nu_mns]
+    quant_names = ["R", "Z", "|B|", "sqrt(g)_B", "nu"]
+    quant_parity = ["cos", "sin", "cos", "cos", "sin"]
+
+    xm = file.variables["ixm_b"][:].filled()
+    xn = file.variables["ixn_b"][:].filled()
+
+    # load in the .nc from the cpp version
+
+    file_cpp = Dataset(Cpp_booz_output_path, mode="r")
+
+    R_mnc_cpp = file_cpp.variables["rmnc_b"][:].filled()
+    Z_mns_cpp = file_cpp.variables["zmns_b"][:].filled()
+    B_mnc_cpp = file_cpp.variables["bmnc_b"][:].filled()
+    nu_mns_cpp = file_cpp.variables["pmns_b"][:].filled()
+    g_mnc_cpp = file_cpp.variables["gmn_b"][:].filled()
+
+    xm_cpp = file_cpp.variables["ixm_b"][:].filled()
+    xn_cpp = file_cpp.variables["ixn_b"][:].filled()
+
+    quantities_cpp = [R_mnc_cpp, Z_mns_cpp, B_mnc_cpp, g_mnc_cpp, nu_mns_cpp]
+
+    np.testing.assert_allclose(xm_cpp, xm, atol=1e-16)
+    np.testing.assert_allclose(xn_cpp, xn, atol=1e-16)
+
+    # precompute some things to avoid redundant calculations
+    quant_mns = {}
+    quant_cpp_mns = {}
+    ms = {}
+    ns = {}
+    quant_bases = {}
+    quant_mode_idxs = {}
+    for quant_index in range(len(quantities)):
```

Comment:
this test feels overly complicated, shouldn't we be able to directly compare arrays from the 2 `.nc` files?